src/ansi-c/library/string.c

```diff
@@ -52,15 +52,14 @@ __inline char *__builtin___strcat_chk(char *dst, const char *src, __CPROVER_size
   while(dst[i]!=0) i++;
 
   __CPROVER_size_t j=0;
-  char ch;
-  do
+  char ch = 1;
+  for(; i < s && ch != (char)0; ++j)
   {
```

Comment:
It's a matter of beauty, but why not increase the 'i' instead of the 'j' in the 'for'? Or both?

Reply:
I'll do both!